tests/Psl/Type/ArrayKeyTypeTest.php

```diff
@@ -21,12 +21,12 @@ public function getValidCoercions(): iterable
         yield ['123', '123'];
         yield ['1e23', '1e23'];
         yield [$this->stringable('123'), '123'];
-        yield [1.0, 1];
+        yield [1.0, '1.0'];
```

Comment:
This might be a BC break?

Reply:
hm, yes, and i don't think we should do this for array keys, we should probably flip the array key union type to check  for int first, then string.

Reply:
In that case, line 21 will return an integer instead of a string. Which is also not correct :)

Reply:
It shouldn't, array-key is a union of `string` and `int` type, and union type should always try to assert that the value is one of either types before trying do any coercing, so if line 21 fails, it's a bug. https://github.com/azjezz/psl/blob/1.5.x/src/Psl/Type/Internal/UnionType.php#L69-L75

Reply:
I'm sorry, it is the one on line 23 : a stringable object that returns a numeric string will return an int.
Since the integer type is also coercing stringables.

Reply:
Hm, maybe we shouldn't extend union type for array-key then ( this is actually the exact same problem why i didn't finish https://github.com/php/php-src/pull/4073, i didn't know how to handle this case 😆 )

Reply:
also, php doesn't support stringable -> int, and i don't think we should either: https://3v4l.org/0G6St